core/ibft.go

```diff
@@ -328,6 +328,7 @@ func (i *IBFT) RunSequence(ctx context.Context, h uint64) {
 			i.moveToNewRound(ev.round)
 			i.acceptProposal(ev.proposalMessage)
 			i.state.setRoundStarted(true)
+			i.sendPrepareMessage(view)
```

Comment:
You have not included a check for the sent out prepare to an existing test that covers this scenario:
https://github.com/0xPolygon/go-ibft/blob/21fc3f6b02b03a5351134bc4ffaa1a8967df6a31/core/ibft_test.go#L2335-L2337

The easiest way to add it would be to define the transport method for the mock, and compare the data from the multicasted prepare

Reply:
Resolving as @trimixlover mentioned it will be tackled in a separate PR